debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/events/Transaction.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.oracle.logminer.events;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+
+import io.debezium.connector.oracle.Scn;
+import io.debezium.connector.oracle.logminer.processor.infinispan.VisibleForMarshalling;
+
+/**
+ * A logical database transaction
+ *
+ * @author Chris Cranford
+ */
+public class Transaction {
+
+    private String transactionId;
+    private Scn startScn;
+    private Instant changeTime;
+    private List<LogMinerEvent> events;
+    private Set<Long> hashes;
+
+    @VisibleForMarshalling
+    public Transaction(String transactionId, Scn startScn, Instant changeTime, List<LogMinerEvent> events, Set<Long> hashes) {
+        this.transactionId = transactionId;
+        this.startScn = startScn;
+        this.changeTime = changeTime;
+        this.events = events;
+        this.hashes = hashes;
+    }
+
+    public Transaction(String transactionId, Scn startScn, Instant changeTime) {
+        this(transactionId, startScn, changeTime, new ArrayList<>(), new HashSet<>());
+    }
+
+    public String getTransactionId() {
+        return transactionId;
+    }
+
+    public Scn getStartScn() {
+        return startScn;
+    }
+
+    public Instant getChangeTime() {
+        return changeTime;
+    }
+
+    public List<LogMinerEvent> getEvents() {
+        return events;
+    }
+
+    public Set<Long> getHashes() {
```

Comment:
The collections would need to be exposed for marshalling purposes with Infinispan, but I could annotate the methods with the specific visibility annotation indicating why so we know not to use those.  Does that seem acceptable to you?

Reply:
Yes, sounds good

Reply:
While I think it would be nice to not expose these collections, I think doing so would be a pretty hefty change to make that I'd prefer to do as a refactor in alpha2.  Ideally I think what we need to do here a few things:

* Fold `TransactionReconcilation` into `Transaction`.  
* Expose a `#commit` method on `Transaction` and let the processors call this method; allowing us to fold a large portion of the processor's `handleCommit` logic into `Transaction`. 
* And the hash ideas you pointed out above.

Reply:
I agree, let's make a follow-up Jira for that.

Reply:
Added as part of https://issues.redhat.com/browse/DBZ-3798